src/main/scala/org/constellation/datastore/SnapshotTrigger.scala

```diff
@@ -35,10 +39,11 @@ class SnapshotTrigger(periodSeconds: Int = 5)(implicit dao: DAO, cluster: Cluste
         elapsed <- IO(System.currentTimeMillis() - startTime)
         _ = logger.debug(s"Attempt snapshot took: $elapsed millis")
         _ <- snapshotResult match {
+          case Left(SnapshotIllegalState) =>
+            handleError(SnapshotIllegalState, stateSet)
+              .flatMap(_ => contextShift.shift >> dao.snapshotBroadcastService.verifyRecentSnapshots())
```

Comment:
What about switching back?

Reply:
switching back to what ?

Reply:
You’re jumping here to the bounded thread pool, aren’t you?

Reply:
Periodic is and upper level guard to evaluate on correct thread and in SnapshotTrigger it's a last operation so there is nothing to execute after